Neos.ESCR.AssetUsage/Classes/Service/GlobalAssetUsageService.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Neos\ESCR\AssetUsage\Service;
+
+use Neos\ContentRepository\Core\ContentRepository;
+use Neos\ContentRepository\Core\Factory\ContentRepositoryId;
+use Neos\ContentRepository\Core\Factory\ContentRepositoryServiceInterface;
+use Neos\ContentRepositoryRegistry\ContentRepositoryRegistry;
+use Neos\ESCR\AssetUsage\AssetUsageFinder;
+use Neos\ESCR\AssetUsage\Dto\AssetUsageFilter;
+use Neos\ESCR\AssetUsage\Dto\AssetUsages;
+use Neos\Flow\Annotations as Flow;
+use Neos\ESCR\AssetUsage\Projection\AssetUsageRepository;
+
+
+class GlobalAssetUsageService implements ContentRepositoryServiceInterface
+{
+    /**
+     * @var array
+     */
+    #[Flow\InjectConfiguration]
+    protected array $flowSettings;
+
+    protected ?array $repositories = null;
+
+    public function __construct(
+        protected readonly ContentRepositoryRegistry $contentRepositoryRegistry,
+    ) {
+    }
+
+    public function findAssetUsageByAssetId(string $assetId): AssetUsages
+    {
+        $assetUsages = $this->withAllRepositories(
+            function (ContentRepository $repository) use ($assetId) {
+                $filter = AssetUsageFilter::create()
+                    ->withAsset($assetId)
+                    ->groupByNode();
+
+                return $repository->projectionState(AssetUsageFinder::class)->findByFilter($filter);
+            }
+        );
+
+        return AssetUsages::fromArrayOfAssetUsages($assetUsages);
+    }
+
+    public function removeAssetUsageByAssetId(string $assetId): void
+    {
+        $this->withAllRepositories(
+            fn(AssetUsageRepository $repository) => $repository->removeAsset($assetId)
+        );
+    }
+
+    private function getContentRepositories(): array
```

Comment:
this is something I am unsure about. Maybe we can configure this per-site? (I guess easier to discuss via video call etc)

Reply:
Actually we discussed a lot, but not this place.  Do we still need to discuss this?